### DISCUSSION TITLE
Chatterino work with youtube streams

### DISCUSSION THREAD

**Discussion by Daldameht** [Chatterino/chatterino2#3298] [Ideas]
Im not sure if chatterino is only meant to be a Twitch only client, but if not, then why no allow chatterino to also work with Youtube chat. Working with the Youtube chat client is unbearable and even though Youtube seems to be changing its UI, the chat is still terrible. Thank you

**Comment by Felanbird** [comment]
We've had the discussion about implementing other services many many times, and while Youtube is the strongest argument since they aren't exactly going to collapse tomorrow, they still do not use IRC for their chat which is what this entire application revolves around.

Basically there's a very very very low chance of this, maybe if forsen ever moves to Youtube and the chat pushes someone to the brink of insanity.

`note:` try [7tv](https://7tv.app) - I know they've been working on trying to make Youtube chat less terrible.

**Reply by Daldameht** [reply]
I have, 7tv does have youtube support. I just wanted to see if I can get the all in one chat experience for both platforms, thanks anyway

**Reply by Mm2PL** [reply]
The closest thing I can see happening to YouTube chat getting added to Chatterino is an IRC <->YouTube proxy getting made. 

###### I'm not crazy enough to do this.

**Reply by ALazyMeme** [reply]
> The closest thing I can see happening to YouTube chat getting added to Chatterino is an IRC <->YouTube proxy getting made.
> 
> ###### I'm not crazy enough to do this.

This what you're looking for? https://irc.septapus.com/

**Comment by dCo3lh0** [comment]
Can i add a sugestion, maybe its not used much compared to twitch, so isnt worth it, but how about dlive.tv and trovo.live? Maybe Odysee which is growing by the day?

Edit: Maybe one more service, Picarto.tv, its the artist streaming service, for graphic artists, designers and such. They mainly already use twitch, but there is artists using these services as well.

**Reply by LosFarmosCTL** [reply]
Same arguments as with youtube apply, platform implementations are fundamentally different, chat is either not available at all for 3rd parties, or not via IRC, APIs are completely different. Supporting another platform/chat protocol would be basically a complete rewrite at that point, no reason for chatterino to implement any of that, not even for a platform like youtube, and especially not for some small platforms that reach a negligible user base compared to twitch.

**Reply by ALazyMeme** [reply]
Trovo's chat is websocket based it seems: https://developer.trovo.live/docs/Chat%20Service.html
Seems like dlive uses irc, so that can be added via the irc feature in chatterino. Just wouldn't have any emotes.

**Comment by ashugg** [comment]
Chatterino fork [youtterino](https://github.com/TheSomethingOrg/youtterino) looks like it was created with the intent to bridge this gap, but there have been no releases yet

Actually, only one commit, so maybe that person is waiting to be filled with divine inspiration or something 😄

**Comment by Nerixyz** [comment]
As explained in https://github.com/Chatterino/chatterino2/discussions/3298#discussioncomment-1505818, Chatterino getting native support for YouTube won't happen in the foreseeable future. Can't find a comment, but it has been said before that Chatterino is primarily a Twitch chat client.

However, we do have a plugin system. My goal is to make that powerful enough to provide an interface for custom channels. Not only for YouTube but also (potentially) for StreamElements/-Labs, Kick, TikTok and similar (as there have been requests for these platforms too). Of course, plugins won't be able to achieve a Twitch-like experience, but I do think that we can get 70-90%. An API for plugins to say "I can manage a channel for platform X and I need one text input" shouldn't be too far away (hopefully). If you have more ideas, take a look at https://github.com/Chatterino/chatterino2/discussions/4999.

There's also an implementation of rendering/creating messages with plugins from YouTube live streams in https://github.com/Remahy/Chatterino-Plugins/tree/5e28bd83ef6a7d5eec36202cc7517193709eb1b8/yt-chat.
